### ISSUE TITLE
watch not working when creating charts or maps

### ISSUE THREAD

**Issue by nshiab** [nshiab/simple-data-analysis#1191]
# deno run --watch fails to restart process after writeChart or writeMap

**Context/Background:**
The `watch` mode in Deno (`deno run --watch`) stops working after a chart or map is generated using `writeChart` or `writeMap`. The script executes once, but subsequent file changes do not trigger a restart, requiring a manual rerun.

**Description/Steps to Reproduce:**
1. Create a script `test.ts` that loads data and calls `table.writeChart(...)`.
2. Ensure `await sdb.done()` is called at the end.
3. Run the script: `deno run -A --watch test.ts`.
4. The script executes and produces the chart.
5. Modify `test.ts` and save.
6. **Observed Behavior:** The process remains idle; Deno does not restart the script.
7. **Expected Behavior:** Deno should detect the change and restart the script automatically.

**Acceptance Criteria:**
- [ ] Deno successfully restarts the script on file changes after `writeChart`/`writeMap` calls.
- [ ] No lingering handles or resources from `journalism-dataviz` prevent the process from resetting.
- [ ] Added a regression test (or an example script) to verify `watch` behavior doesn't break again.

**Data Shapes & Signatures:**
No changes to existing signatures are expected, but internal cleanup in `journalism-dataviz` may be required.
```ts
// Target methods
await table.writeChart(chartFunction, outputPath);
await table.writeMap(mapFunction, outputPath);
```

**Technical Notes:**
- The issue is specific to the charting methods. Commenting out `writeChart` allows `watch` to work normally.
- `journalism-dataviz` uses `linkedom` for DOM shimming and `@resvg/resvg-js` for PNG rendering.
- Deno's watcher can be sensitive to open resources (e.g., workers, timers, or native library handles).
- Need to investigate if `Resvg` or `linkedom` require explicit disposal or if the global shim cleanup in `saveChart.ts` is incomplete.

_Edit (2026-05-08): Refined description based on discussion. Confirmed the issue is isolated to charting methods and affects Deno's watcher restart mechanism._
